### PR TITLE
First steps

### PR DESCRIPTION
**Pull request by maxxxxxdlp**
- Improve development documentation
- Remove demo files
- Design (tentative) app logo
- Fix errors in manifest.json to make Google Chrome accept as as an extension
- Fix webpack source map not getting loaded in development
- Render a basic clickable button. The "Click Me" button is now added to the Google Chrome toolbar!

### CODE REVIEW

File: src/src/components/App/index.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { commonText } from '../../localization/common';
+import { useBooleanState } from '../../hooks/useBooleanState';
+import { Portal } from '../Molecules/Portal';
+
+export function App() {
+  const [isOpen, _, handleClose, handleToggle] = useBooleanState();
```
Comment (jetsemr) [inline src/src/components/App/index.tsx:7]: What is useBooleanState? Why not use useState react hook?

File: src/src/components/Molecules/Portal.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
```
Comment (jetsemr) [inline src/src/components/Molecules/Portal.tsx:1]: Not a big deal, but I feel like it can be helpful for readability l to put React methods directly here instead of throughout the file using React.useEffect() etc.

File: src/src/components/Molecules/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Molecules
```
Comment (jetsemr) [inline src/src/components/Molecules/README.md:1]: What is Molecules?

File: src/src/css/main.css [hunk 4]
```diff
@@ -6,150 +6,8 @@
 
 @layer base {
```
Comment (jetsemr) [inline src/src/css/main.css:7]: Might be useful for us to set up a group meeting and detail out styling for the app.

File: src/src/components/App/index.tsx [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (jetsemr) [inline src/src/components/App/index.tsx:1]: Why is src/public/index.html getting deleted? Sorry didn't know where else to comment this.

File: src/src/index.tsx [hunk 6]
```diff
@@ -3,14 +3,21 @@ import './css/main.css';
...
-const root = ReactDOM.createRoot(document.getElementById('root')!);
-root.render(
-  <React.StrictMode>
-    <Contexts>
-      <App/>
-    </Contexts>
-  </React.StrictMode>
-);
+const container =
+  document.querySelector('header div[aria-label="Search"]')?.parentElement
```
Comment (Durbatuluk1701) [inline src/src/index.tsx:10]: Could we maybe leave a comment describing what this `container` is ultimately supposed to be referring to?

**Review by Durbatuluk1701 — APPROVED** [review]
Think it looks good other than the unresolved comments. Also, works to run for me locally

File: src/src/components/App/index.tsx (hunk 1 above)
Comment (maxxxxxdlp) [reply src/src/components/App/index.tsx:7]: I explained my reasoning in the documentation for that hook - https://github.com/maxxxxxdlp/calendar-stats/blob/1c4555747c45cb32c60907742aaa00faed188349/src/src/hooks/useBooleanState.tsx#L4-L36


**TLDR:** I find `onClick={handleClose}` easier to read than `onClick={(): void => setIsOpen(false)}`

But, if you think I should keep it simple I can revert it to `React.useState<boolean>();`

File: src/src/components/Molecules/Portal.tsx (hunk 2 above)
Comment (maxxxxxdlp) [reply src/src/components/Molecules/Portal.tsx:1]: Interesting, can you explain more why you think so?

I prefer `React.useEffect()` for readability as it makes it clear in code that the hook is comming from react, as opposed to a user hook.

Basically, using the `React.` namespace makes it easy to differentiate default react hooks from my custom hooks.

File: src/src/components/App/index.tsx (hunk 1 above)
Comment (jetsemr) [reply src/src/components/App/index.tsx:7]: All good appreciate the explanation!

File: src/src/components/Molecules/README.md (hunk 3 above)
Comment (maxxxxxdlp) [reply src/src/components/Molecules/README.md:1]: I have a link to the Atomic Design principle in that file - https://github.com/maxxxxxdlp/calendar-stats/tree/first-steps/src/src/components/Molecules

TLDR: `Atoms` are the most basic building blocks of the UI. `Molecules` are a bit higher level than that and can combine multiple `Atoms`.

File: src/src/css/main.css (hunk 4 above)
Comment (maxxxxxdlp) [reply src/src/css/main.css:7]: One approach is to try and make our extension follow Google style language so as to fit in with the rest of the UI elements in the Google Calendar

File: src/src/components/Molecules/README.md (hunk 3 above)
Comment (jetsemr) [reply src/src/components/Molecules/README.md:1]: Interesting - seems cool!

File: src/src/css/main.css (hunk 4 above)
Comment (jetsemr) [reply src/src/css/main.css:7]: I like this, will look into it more because I am not familiar.

File: src/src/components/App/index.tsx (hunk 5 above)
Comment (jetsemr) [reply src/src/components/App/index.tsx:1]: I see, makes sense.

**Review by jetsemr — APPROVED** [review]
Looks good!

File: src/src/index.tsx (hunk 6 above)
Comment (maxxxxxdlp) [reply src/src/index.tsx:17]: Good idea! Added in [fafba95](https://github.com/maxxxxxdlp/calendar-stats/pull/27/commits/fafba9599634c8637a73d17191c0f44691a3f30c)
